src/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -20,6 +20,18 @@ public static void NoSetter()
             Assert.Equal("DefaultValue", objCopy.MyString);
         }
 
+        [Fact]
+        public static void IgnoreReadOnlyProperties()
+        {
+            var options = new JsonSerializerOptions();
+            options.IgnoreReadOnlyProperties = true;
+
+            var obj = new ClassWithNoSetter();
+
+            string json = JsonSerializer.ToString(obj, options);
+            Assert.Equal(@"{}", json);
```

Comment:
I think all the "basic" tests should contain more complex JSON cases, which should included nested types:
[]
{}
[{}]
{"foo":[]}
[1, {}]
{"1": 1, "foo":[]}
[1, {}, 2]
{"1": 1, "foo":[], "2": 2}

And then inject readonly properties, or null properties everywhere possible in the above cases.
`+` A mode with comments everywhere.

Reply:
There are many tests like above in other areas. For PropertyVisibilityTests.cs I am just setting a subset.

However I did add a couple tests based on what you had above. Thanks